File: Evaluate Reverse Polish Notation.js
* @param {string[]} tokens
* @return {number}
*/
// var evalRPN = function(tokens) {
//     let stack = [];
//     for (let token of tokens){
//         if (!['+','-','*','/'].includes(token)){
//             stack.push(parseInt(token));
//         }
//         else{
//             let right = stack.pop() , left = stack.pop()
//             if (token==='+'){
//                 stack.push(right+left);
//             }
//             else if (token==='-'){
//                 stack.push(left-right);
//             }
//             else if (token==='*'){
//                 stack.push(left*right);
//             }
//             else if (token==='/'){
//                 stack.push(parseInt(left/right));
//             }
//         }
//     }
//     return stack.pop();
// };
var evalRPN = function(tokens) {
    let stack = [];
    for (let token of tokens){
        switch(token) {
            case '+':
                stack.push(stack.pop() + stack.pop());
                break;
            case '-':
                stack.push(-stack.pop() + stack.pop());
                break;
            case '*':
                stack.push(stack.pop() * stack.pop());
                break;
            case '/':
                let right = stack.pop(), left = stack.pop();
                stack.push((left / right) | 0); // bitwise OR for truncating towards zero
                break;
            default:
                stack.push(Number(token)); // Direct conversion to Number
        }
    }
    return stack.pop();
};

File: Evaluate Reverse Polish Notation.py
def evalRPN(self, tokens: List[str]) -> int:
    stack = []
    for token in tokens:
        if token not in ['+','-','/','*']:
            stack.append(int(token))
        else:
            right , left = stack.pop(), stack.pop()
            if token=='+':
                stack.append(right+left)
            elif token =='-':
                stack.append(left - right)
            elif token=='*':
                stack.append(left * right)
            elif token=='/':
                stack.append(int(left/right))
    return stack.pop()